specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx

```diff
@@ -152,10 +154,15 @@ export function PrepDialog({
       <Form
         id={id('form')}
         onSubmit={(): void => {
-          const itemTable = strictGetTable(
-            `${table.name}Preparation`
-          ) as SpecifyTable<
-            DisposalPreparation | GiftPreparation | LoanPreparation
+          const tableName =
+            table.name === 'ExchangeOut'
+              ? `${table.name}Prep`
+              : `${table.name}Preparation`;
+          const itemTable = strictGetTable(tableName) as SpecifyTable<
+            | DisposalPreparation
+            | ExchangeOutPrep
+            | GiftPreparation
+            | LoanPreparation
           >;
```

Comment:
Instead of hardcoding the table name here, it can be fetched dynamically using the constant defined in an above comment (a tuple containing the name of all InteractionPreparations): 

```suggestion
          const itemTable = defined(
            table.relationships.find((relationship) =>
              interactionPrepTables.includes(relationship.relatedTable.name)
            )?.table
          ) as SpecifyTable<AnyInteractionPreparation>;
```